### PR TITLE
[quant] Add quantizer class skeleton

### PR DESCRIPTION
**Pull request by asl3**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #79845
* #79940
* #79530
* #79499
* __->__ #79936

### Summary:
This PR adds a quantizer class skeleton to store APoT quantize and dequantize methods, as well as APoT quantization parameters.

### Test Plan:
Run unit tests with: `python pytorch/test/quantization/core/experimental/test_quantizer.py`

### CODE REVIEW

**Comment by dzdang** [comment]
PR name and summar yshould be changed

File: torch/ao/quantization/experimental/quantizer.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import torch
+from torch import Tensor
+
+# class to store APoT quantizer
+# implements quantize and dequantize
+# and stores all quantization parameters
+class APoTQuantizer(torch.Tensor):
```
Comment (jerryzh168) [inline torch/ao/quantization/experimental/quantizer.py:7]: nit: quantizer doesn't need to inherit from Tensor, it's a separate object

**Review by jerryzh168 — APPROVED** [review]
looks good, had one comment inline

**Comment by asl3** [comment]
git merge -g

**Pull request closed without merging**